[easy] Don't check uint32 >= 0

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #109933
* __->__ #109932

This check was triggering a log line for each of the 1600 files that get compiled when you modify ivalue_inl.h. Remove this check to silence the noisy compiler warning; uint32 will always be >= 0.

davidberard98 committed:
[easy] Don't check uint32 >= 0

This check was triggering a log line for each of the 1600 files that get compiled when you modify ivalue_inl.h. Remove this check to silence the noisy compiler warning.

[ghstack-poisoned]

vadimkantorov commented on aten/src/ATen/core/ivalue.h:
```diff
@@ -1281,7 +1281,7 @@ struct TORCH_API IValue final {
 
     TORCH_INTERNAL_ASSERT_DEBUG_ONLY(
-        uint32_t(tag) >= 0 && uint32_t(tag) < kNumTags,
+        uint32_t(tag) && uint32_t(tag) < kNumTags,
```
I think this is slightly not equivalent to the previous code.
`uint32_t(tag) >= 0 ` accepted the value `tag == 0` and the new condition `uint32_t(tag) &&` seems to include it, right?

If this case is not needed, should `uint32_t(tag) && ` be dropped as well from the check?

vadimkantorov replied:
Or if really a check was meant, it should be instead `tag >= 0`...

davidberard98 replied:
yeah you're right

davidberard98 committed:
Update on "[easy] Don't check uint32 >= 0"


This check was triggering a log line for each of the 1600 files that get compiled when you modify ivalue_inl.h. Remove this check to silence the noisy compiler warning; uint32 will always be >= 0.

[ghstack-poisoned]

davidberard98 committed:
Update on "[easy] Don't check uint32 >= 0"


This check was triggering a log line for each of the 1600 files that get compiled when you modify ivalue_inl.h. Remove this check to silence the noisy compiler warning; uint32 will always be >= 0.

[ghstack-poisoned]

davidberard98 committed:
Update on "[easy] Don't check uint32 >= 0"


This check was triggering a log line for each of the 1600 files that get compiled when you modify ivalue_inl.h. Remove this check to silence the noisy compiler warning; uint32 will always be >= 0.

[ghstack-poisoned]
